### PR TITLE
Update dependencies and SDK

### PR DESCRIPTION
**Pull request by natemcmaster**
Changes:
* Update .NET Core SDK to latest preview 3
* Manually update corefx dependencies to match core-setup
~~* Add `darc verify` to PR checks~~

cc @bricelam

### CODE REVIEW

**Comment by rynowak** [comment]
Should we do a similar for tooling?

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -80,9 +76,9 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>03497f16b93264158c28cee4eea767b53bc750f5</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.Platforms" Version="3.0.0-preview.19105.1">
+    <Dependency Name="Microsoft.NETCore.Platforms" Version="3.0.0-preview.19109.6">
```
Comment (dougbu) [inline eng/Version.Details.xml]: To confirm: None of the other toolsets have newer versions?

**Comment by natemcmaster** [comment]
Yes. Darc will not automatically update your .NET Core SDK version, so if you want to build with the latest and greatest, update your SDK version in global.json manually.

File: eng/Version.Details.xml (hunk 1 above)
Comment (natemcmaster) [reply eng/Version.Details.xml]: They might, but the bot is working correctly for those. This one is currently busted due to some build issues in the corefx repo.

**Pull request merged**